references/embedding/train.py

```diff
@@ -0,0 +1,149 @@
+import os
+
+import torch
+from torch.optim import Adam
+from torch.utils.data import DataLoader
+from torch.utils.data.dataset import Subset
+from torch.nn.modules.distance import PairwiseDistance
+
+import torchvision.transforms as transforms
+from torchvision.datasets import FashionMNIST
+
+from loss import TripletMarginLoss
+from sampler import PKSampler
+from model import EmbeddingNet
+
+
+def train_epoch(model, optimizer, criterion, data_loader, device, epoch, print_freq):
+    model.train()
+    running_loss = 0
+    running_frac_pos_triplets = 0
+    for i, data in enumerate(data_loader):
+        optimizer.zero_grad()
+        samples, targets = data[0].to(device), data[1].to(device)
+
+        embeddings = model(samples)
+
+        loss, frac_pos_triplets = criterion(embeddings, targets)
+        loss.backward()
+        optimizer.step()
+
+        running_loss += loss.item()
+        running_frac_pos_triplets += float(frac_pos_triplets)
+
+        if i % print_freq == print_freq - 1:
+            i += 1
+            avg_loss = running_loss / print_freq
+            avg_trip = 100.0 * running_frac_pos_triplets / print_freq
+            print('[{:d}, {:d}] | loss: {:.4f} | % avg hard triplets: {:.2f}%'.format(epoch, i, avg_loss, avg_trip))
+            running_loss = 0
+            running_frac_pos_triplets = 0
+
+
+@torch.no_grad()
+def evaluate(model, dataset, device):
+    model.eval()
+    embeds, labels = None, None
+    dists, targets = None, None
+
+    loader = DataLoader(dataset, batch_size=512, shuffle=False, num_workers=4)
```

Comment:
Why not pass the dataloader as an argument to the function?
This keeps the `train` and `evaluate` functions consistent

Reply:
I was going to do this, but I forgot about it before sending the PR. This has been changed.